flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/statistics/OnlineHistogram.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.statistics
+
+/** Base trait for an Online Histogram
```

Comment:
An Online Histogram is meant to approximate a data set with a distribution. 
So, for example, for discrete valued data, we store counters of every class.
For continuous data, we learn a distribution on a data set as more and more elements come along.

It is online in the sense that we don't require the whole data set to build it. It is built incrementally, and for two parts of the data set, it can be merged to provide statistics for the combined set.